File: java/eyes.playwright.java/src/main/java/com/applitools/eyes/playwright/locators/OcrRegion.java
package com.applitools.eyes.playwright.locators;

import com.applitools.eyes.Region;
import com.applitools.eyes.locators.BaseOcrRegion;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.microsoft.playwright.ElementHandle;
import com.microsoft.playwright.Locator;

public class OcrRegion extends BaseOcrRegion {
    @JsonIgnore
    private ElementHandle element;

    @JsonIgnore
    private Locator locator;

    @JsonIgnore
    private String selector;

    @JsonIgnore
    private Region region;

    public OcrRegion(ElementHandle element) {
        this.element = element;
    }

    public OcrRegion(Locator locator) {
        this.locator = locator;
    }

    public OcrRegion(Region region) {
        this.region = region;
    }

    public OcrRegion(String selector) {
        this.selector = selector;
    }

    public ElementHandle getElement() {
        return element;
    }

    public Locator getLocator() {
        return locator;
    }

    public String getSelector() {
        return selector;
    }

    public Region getRegion() {
        return region;
    }
}

File: dotnet/Eyes.Playwright.DotNet/Locators/OcrRegion.cs
using Applitools.Utils.Geometry;
using Microsoft.Playwright;

namespace Applitools.Playwright
{
    public class OcrRegion : OcrRegionBase
    {
        private IElementHandle element_;

        private ILocator locator_;

        private string selector_;

        public OcrRegion(IElementHandle element)
        {
            element_ = element;
        }

        public OcrRegion(ILocator locator)
        {
            locator_ = locator;
        }

        public OcrRegion(Region region)
        {
            Region(region);
        }

        public OcrRegion(string selector)
        {
            selector_ = selector;
        }
        
        public OcrRegion Element(IElementHandle element)
        {
            element_ = element;
            return this;
        }

        internal IElementHandle GetElement()
        {
            return element_;
        }
        
        public OcrRegion Locator(ILocator locator)
        {
            locator_ = locator;
            return this;
        }

        internal ILocator GetLocator()
        {
            return locator_;
        }
        
        public OcrRegion Selector(string selector)
        {
            selector_ = selector;
            return this;
        }

        internal string GetSelector()
        {
            return selector_;
        }
        
        public new OcrRegion Region(Region? region)
        {
            base.Region(region);
            return this;
        }
    }
}